### ISSUE TITLE
python39 can't be installed

### ISSUE THREAD

**Issue by KCynk** [NixOS/nixpkgs#295986]
### Describe the bug
python39 can't be installed

### Steps To Reproduce
1. sudo vim /etc/nixos/configuration.nix
1. add python39 to system packages section 
environment.systemPackages = with pkgs; [
   python39
];
3. sudo nixos-rebuild switch
4. Error after executing command

### Additional context
```
[kamilcynk@nixos:~]$ sudo nixos-rebuild switch 
building Nix...
building the system configuration...
these 13 derivations will be built:
  /nix/store/1vij562n5jjgkfnq5a6wka3qkxid2y72-python3.9-3.9.18-doc.drv
  /nix/store/sci64ads4jnw3d7isag5i13bakl4a3x4-system-path.drv
  /nix/store/y07mm96dy4p709rlvac31jrxyiy9wcxv-dbus-1.drv
  /nix/store/qp7vlzlm3nkq9bdfr1f1cdrflswwnk22-X-Restart-Triggers-dbus.drv
  /nix/store/167bn46nfypvlr3w8zcd84qv44v7d5nw-unit-dbus.service.drv
  /nix/store/1hlbj6ap2l034mqk1pc4i3phsfgnvva4-X-Restart-Triggers-polkit.drv
  /nix/store/j9xr88z3w7ad38vhvib6p0cvahi3kpk9-unit-polkit.service.drv
  /nix/store/msm9gpwjxkafqx9lr6bfws8aiwds1g0v-unit-accounts-daemon.service.drv
  /nix/store/jgyzp6lpcyjwlvvc34nn43gy4a91bwv9-system-units.drv
  /nix/store/qi95lxmip2j58i9s99mf9356yhj4j39j-unit-dbus.service.drv
  /nix/store/zqvf47j8lzglhdp9vnnlzq77y4dlcc9n-user-units.drv
  /nix/store/r9imdy1wgcmvdp71bd4073byvxk0x9bq-etc.drv
  /nix/store/i2f2hr5pf1f0hq6kgvvqdxv54agp4d9r-nixos-system-nixos-24.05pre594677.3030f185ba6a.drv
these 29 paths will be fetched (58.89 MiB download, 184.10 MiB unpacked):
  /nix/store/mqm1nwm3dj4lwq83ss8cw0n1hs6yc38z-Python-3.9.18.tar.xz
  /nix/store/a49ysmb5g5gri312p47rwssdwbnq39mc-python3-3.9.18
  /nix/store/7xqgvcyvn79nls466lxqb9pv3vrmjq70-python3.11-alabaster-0.7.13
  /nix/store/591zy6brzbbsd3r2p3lv05pvbxf9h2gx-python3.11-babel-2.14.0
  /nix/store/k32vjcpvm332xh07092lwshhp1ag5p57-python3.11-brotlicffi-1.1.0.0
  /nix/store/xv6rsgaw6iqyps2kyzxzashrdi99q5ar-python3.11-certifi-2023.11.17
  /nix/store/38q7flpba3qvndx07nc6z3hdhbq3bfsa-python3.11-cffi-1.16.0
  /nix/store/bc34adw3kpbrn6sv7z8dvp4yp8hx5cfj-python3.11-charset-normalizer-3.3.2
  /nix/store/mfihnifp4kcydmjasxjmcjwk6pz3lq4k-python3.11-docutils-0.20.1
  /nix/store/d6z411zplz45n7zchsjn4kaddbpspz6a-python3.11-idna-3.6
  /nix/store/n6cd6m3qsl6phqxs256wx8h82xy7qyif-python3.11-imagesize-1.4.1
  /nix/store/2ldkzb65dnqxlp4jd4kfqi3k3rd657x1-python3.11-jinja2-3.1.3
  /nix/store/r5rvnih8bmkypzxwq8p2v3lvd40f8z9q-python3.11-markupsafe-2.1.5
  /nix/store/vl4i7c209j460iyhdxabzsv6qwb95gqj-python3.11-pycparser-2.21
  /nix/store/b1v8614b2llcfdamwmk3ysjrvw46ihlm-python3.11-pygments-2.17.2
  /nix/store/r3d92pnjsw93b0ng68nipbzp2xkzsv5a-python3.11-pystemmer-2.2.0
  /nix/store/12z6gkpsv77kdpia2rcsdzddhq6iym0d-python3.11-python-docs-theme-2024.2
  /nix/store/k0pnika5g2lqdxl12yffwydp4v8z8b7n-python3.11-requests-2.31.0
  /nix/store/af9kdcf7h65h459dgbxvdqzk1whl7gq8-python3.11-snowballstemmer-2.2.0
  /nix/store/7z8m73ixz3sw59df6rxczzzqg3jhmgsd-python3.11-sphinx-7.2.6
  /nix/store/21sih7jyyl5s6wbybf6p8nb15br2h3hw-python3.11-sphinx-hook
  /nix/store/gc4v11lkb5fa9f0d0lp8w82w1wchga4m-python3.11-sphinxcontrib-applehelp-1.0.7
  /nix/store/ijvpvpgm697gaqimd2iv2wrgnw22p7hq-python3.11-sphinxcontrib-devhelp-1.0.5
  /nix/store/9akwsv8abxqsifvpp4gadpsila5llj07-python3.11-sphinxcontrib-htmlhelp-2.0.4
  /nix/store/sh4ack2nw58ck75g444hls2xb1q9b708-python3.11-sphinxcontrib-jsmath-1.0.1
  /nix/store/6b41zn1d4vy8qrr516zz87q6m756bdg7-python3.11-sphinxcontrib-qthelp-1.0.6
  /nix/store/2s6sk4qcbaipc7snxv53n8lxxaa9dwgq-python3.11-sphinxcontrib-serializinghtml-1.1.9
  /nix/store/3d840y3v4f9f9vhy6i6mkvj5kqxi8pbn-python3.11-sphinxcontrib-websupport-1.2.6
  /nix/store/3wknzq0y5blgnzynkavnxmai5shy8d2k-python3.11-urllib3-2.1.0
copying path '/nix/store/a49ysmb5g5gri312p47rwssdwbnq39mc-python3-3.9.18' from 'https://cache.nixos.org'...
copying path '/nix/store/mqm1nwm3dj4lwq83ss8cw0n1hs6yc38z-Python-3.9.18.tar.xz' from 'https://cache.nixos.org'...
copying path '/nix/store/591zy6brzbbsd3r2p3lv05pvbxf9h2gx-python3.11-babel-2.14.0' from 'https://cache.nixos.org'...
copying path '/nix/store/xv6rsgaw6iqyps2kyzxzashrdi99q5ar-python3.11-certifi-2023.11.17' from 'https://cache.nixos.org'...
copying path '/nix/store/bc34adw3kpbrn6sv7z8dvp4yp8hx5cfj-python3.11-charset-normalizer-3.3.2' from 'https://cache.nixos.org'...
copying path '/nix/store/r5rvnih8bmkypzxwq8p2v3lvd40f8z9q-python3.11-markupsafe-2.1.5' from 'https://cache.nixos.org'...
copying path '/nix/store/vl4i7c209j460iyhdxabzsv6qwb95gqj-python3.11-pycparser-2.21' from 'https://cache.nixos.org'...
copying path '/nix/store/b1v8614b2llcfdamwmk3ysjrvw46ihlm-python3.11-pygments-2.17.2' from 'https://cache.nixos.org'...
copying path '/nix/store/ijvpvpgm697gaqimd2iv2wrgnw22p7hq-python3.11-sphinxcontrib-devhelp-1.0.5' from 'https://cache.nixos.org'...
copying path '/nix/store/gc4v11lkb5fa9f0d0lp8w82w1wchga4m-python3.11-sphinxcontrib-applehelp-1.0.7' from 'https://cache.nixos.org'...
copying path '/nix/store/mfihnifp4kcydmjasxjmcjwk6pz3lq4k-python3.11-docutils-0.20.1' from 'https://cache.nixos.org'...
copying path '/nix/store/d6z411zplz45n7zchsjn4kaddbpspz6a-python3.11-idna-3.6' from 'https://cache.nixos.org'...
copying path '/nix/store/r3d92pnjsw93b0ng68nipbzp2xkzsv5a-python3.11-pystemmer-2.2.0' from 'https://cache.nixos.org'...
copying path '/nix/store/n6cd6m3qsl6phqxs256wx8h82xy7qyif-python3.11-imagesize-1.4.1' from 'https://cache.nixos.org'...
copying path '/nix/store/9akwsv8abxqsifvpp4gadpsila5llj07-python3.11-sphinxcontrib-htmlhelp-2.0.4' from 'https://cache.nixos.org'...
copying path '/nix/store/sh4ack2nw58ck75g444hls2xb1q9b708-python3.11-sphinxcontrib-jsmath-1.0.1' from 'https://cache.nixos.org'...
copying path '/nix/store/6b41zn1d4vy8qrr516zz87q6m756bdg7-python3.11-sphinxcontrib-qthelp-1.0.6' from 'https://cache.nixos.org'...
copying path '/nix/store/2s6sk4qcbaipc7snxv53n8lxxaa9dwgq-python3.11-sphinxcontrib-serializinghtml-1.1.9' from 'https://cache.nixos.org'...
copying path '/nix/store/3wknzq0y5blgnzynkavnxmai5shy8d2k-python3.11-urllib3-2.1.0' from 'https://cache.nixos.org'...
copying path '/nix/store/2ldkzb65dnqxlp4jd4kfqi3k3rd657x1-python3.11-jinja2-3.1.3' from 'https://cache.nixos.org'...
copying path '/nix/store/af9kdcf7h65h459dgbxvdqzk1whl7gq8-python3.11-snowballstemmer-2.2.0' from 'https://cache.nixos.org'...
copying path '/nix/store/38q7flpba3qvndx07nc6z3hdhbq3bfsa-python3.11-cffi-1.16.0' from 'https://cache.nixos.org'...
copying path '/nix/store/3d840y3v4f9f9vhy6i6mkvj5kqxi8pbn-python3.11-sphinxcontrib-websupport-1.2.6' from 'https://cache.nixos.org'...
copying path '/nix/store/k32vjcpvm332xh07092lwshhp1ag5p57-python3.11-brotlicffi-1.1.0.0' from 'https://cache.nixos.org'...
copying path '/nix/store/k0pnika5g2lqdxl12yffwydp4v8z8b7n-python3.11-requests-2.31.0' from 'https://cache.nixos.org'...
copying path '/nix/store/7xqgvcyvn79nls466lxqb9pv3vrmjq70-python3.11-alabaster-0.7.13' from 'https://cache.nixos.org'...
copying path '/nix/store/7z8m73ixz3sw59df6rxczzzqg3jhmgsd-python3.11-sphinx-7.2.6' from 'https://cache.nixos.org'...
copying path '/nix/store/21sih7jyyl5s6wbybf6p8nb15br2h3hw-python3.11-sphinx-hook' from 'https://cache.nixos.org'...
copying path '/nix/store/12z6gkpsv77kdpia2rcsdzddhq6iym0d-python3.11-python-docs-theme-2024.2' from 'https://cache.nixos.org'...
building '/nix/store/1vij562n5jjgkfnq5a6wka3qkxid2y72-python3.9-3.9.18-doc.drv'...
Sourcing sphinx-hook
Running phase: unpackPhase
unpacking source archive /nix/store/mqm1nwm3dj4lwq83ss8cw0n1hs6yc38z-Python-3.9.18.tar.xz
source root is Python-3.9.18
setting SOURCE_DATE_EPOCH to timestamp 1692901018 of file Python-3.9.18/Misc/NEWS
Running phase: patchPhase
Running phase: updateAutotoolsGnuConfigScriptsPhase
Updating Autotools / GNU config script to a newer upstream version: ./config.sub
Updating Autotools / GNU config script to a newer upstream version: ./config.guess
Running phase: installPhase
no Makefile or custom installPhase, doing nothing
Running phase: fixupPhase
Running phase: buildSphinxPhase
Executing buildSphinxPhase
Executing sphinx-build with html builder
Running Sphinx v7.2.6
/build/Python-3.9.18/Doc/tools/extensions/pyspecific.py:30: RemovedInSphinx80Warning: The alias 'sphinx.util.status_iterator' is deprecated, use 'sphinx.util.display.status_iterator' instead. Check CHANGES for Sphinx API modifications.
  from sphinx.util import status_iterator, logging
making output directory... done
locale_dir /build/Python-3.9.18/Doc/locales/en/LC_MESSAGES does not exist
locale_dir /build/Python-3.9.18/Doc/locales/en/LC_MESSAGES does not exist
building [mo]: targets for 0 po files that are out of date
writing output... 
building [html]: targets for 488 source files that are out of date
updating environment: locale_dir /build/Python-3.9.18/Doc/locales/en/LC_MESSAGES does not exist
[new config] 488 added, 0 changed, 0 removed
reading sources... [  0%] about
reading sources... [  0%] bugs
reading sources... [  1%] c-api/abstract
reading sources... [  1%] c-api/allocation
reading sources... [  1%] c-api/apiabiversion
reading sources... [  1%] c-api/arg

Traceback (most recent call last):
  File "/nix/store/7z8m73ixz3sw59df6rxczzzqg3jhmgsd-python3.11-sphinx-7.2.6/lib/python3.11/site-packages/sphinx/cmd/build.py", line 298, in build_main
    app.build(args.force_all, args.filenames)
  File "/nix/store/7z8m73ixz3sw59df6rxczzzqg3jhmgsd-python3.11-sphinx-7.2.6/lib/python3.11/site-packages/sphinx/application.py", line 355, in build
    self.builder.build_update()
  File "/nix/store/7z8m73ixz3sw59df6rxczzzqg3jhmgsd-python3.11-sphinx-7.2.6/lib/python3.11/site-packages/sphinx/builders/__init__.py", line 293, in build_update
    self.build(to_build,
  File "/nix/store/7z8m73ixz3sw59df6rxczzzqg3jhmgsd-python3.11-sphinx-7.2.6/lib/python3.11/site-packages/sphinx/builders/__init__.py", line 313, in build
    updated_docnames = set(self.read())
                           ^^^^^^^^^^^
  File "/nix/store/7z8m73ixz3sw59df6rxczzzqg3jhmgsd-python3.11-sphinx-7.2.6/lib/python3.11/site-packages/sphinx/builders/__init__.py", line 420, in read
    self._read_serial(docnames)
  File "/nix/store/7z8m73ixz3sw59df6rxczzzqg3jhmgsd-python3.11-sphinx-7.2.6/lib/python3.11/site-packages/sphinx/builders/__init__.py", line 441, in _read_serial
    self.read_doc(docname)
  File "/nix/store/7z8m73ixz3sw59df6rxczzzqg3jhmgsd-python3.11-sphinx-7.2.6/lib/python3.11/site-packages/sphinx/builders/__init__.py", line 498, in read_doc
    publisher.publish()
  File "/nix/store/mfihnifp4kcydmjasxjmcjwk6pz3lq4k-python3.11-docutils-0.20.1/lib/python3.11/site-packages/docutils/core.py", line 234, in publish
    self.document = self.reader.read(self.source, self.parser,
                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/nix/store/7z8m73ixz3sw59df6rxczzzqg3jhmgsd-python3.11-sphinx-7.2.6/lib/python3.11/site-packages/sphinx/io.py", line 105, in read
    self.parse()
  File "/nix/store/mfihnifp4kcydmjasxjmcjwk6pz3lq4k-python3.11-docutils-0.20.1/lib/python3.11/site-packages/docutils/readers/__init__.py", line 76, in parse
    self.parser.parse(self.input, document)
  File "/nix/store/7z8m73ixz3sw59df6rxczzzqg3jhmgsd-python3.11-sphinx-7.2.6/lib/python3.11/site-packages/sphinx/parsers.py", line 81, in parse
    self.statemachine.run(inputlines, document, inliner=self.inliner)
  File "/nix/store/mfihnifp4kcydmjasxjmcjwk6pz3lq4k-python3.11-docutils-0.20.1/lib/python3.11/site-packages/docutils/parsers/rst/states.py", line 169, in run
    results = StateMachineWS.run(self, input_lines, input_offset,
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/nix/store/mfihnifp4kcydmjasxjmcjwk6pz3lq4k-python3.11-docutils-0.20.1/lib/python3.11/site-packages/docutils/statemachine.py", line 233, in run
    context, next_state, result = self.check_line(
                                  ^^^^^^^^^^^^^^^^
  File "/nix/store/mfihnifp4kcydmjasxjmcjwk6pz3lq4k-python3.11-docutils-0.20.1/lib/python3.11/site-packages/docutils/statemachine.py", line 445, in check_line
    return method(match, context, next_state)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/nix/store/mfihnifp4kcydmjasxjmcjwk6pz3lq4k-python3.11-docutils-0.20.1/lib/python3.11/site-packages/docutils/parsers/rst/states.py", line 2785, in underline
    self.section(title, source, style, lineno - 1, messages)
  File "/nix/store/mfihnifp4kcydmjasxjmcjwk6pz3lq4k-python3.11-docutils-0.20.1/lib/python3.11/site-packages/docutils/parsers/rst/states.py", line 325, in section
    self.new_subsection(title, lineno, messages)
  File "/nix/store/mfihnifp4kcydmjasxjmcjwk6pz3lq4k-python3.11-docutils-0.20.1/lib/python3.11/site-packages/docutils/parsers/rst/states.py", line 391, in new_subsection
    newabsoffset = self.nested_parse(
                   ^^^^^^^^^^^^^^^^^^
  File "/nix/store/mfihnifp4kcydmjasxjmcjwk6pz3lq4k-python3.11-docutils-0.20.1/lib/python3.11/site-packages/docutils/parsers/rst/states.py", line 279, in nested_parse
    state_machine.run(block, input_offset, memo=self.memo,
  File "/nix/store/mfihnifp4kcydmjasxjmcjwk6pz3lq4k-python3.11-docutils-0.20.1/lib/python3.11/site-packages/docutils/parsers/rst/states.py", line 195, in run
    results = StateMachineWS.run(self, input_lines, input_offset)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/nix/store/mfihnifp4kcydmjasxjmcjwk6pz3lq4k-python3.11-docutils-0.20.1/lib/python3.11/site-packages/docutils/statemachine.py", line 233, in run
    context, next_state, result = self.check_line(
                                  ^^^^^^^^^^^^^^^^
  File "/nix/store/mfihnifp4kcydmjasxjmcjwk6pz3lq4k-python3.11-docutils-0.20.1/lib/python3.11/site-packages/docutils/statemachine.py", line 445, in check_line
    return method(match, context, next_state)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/nix/store/mfihnifp4kcydmjasxjmcjwk6pz3lq4k-python3.11-docutils-0.20.1/lib/python3.11/site-packages/docutils/parsers/rst/states.py", line 3024, in text
    self.section(title.lstrip(), source, style, lineno + 1, messages)
  File "/nix/store/mfihnifp4kcydmjasxjmcjwk6pz3lq4k-python3.11-docutils-0.20.1/lib/python3.11/site-packages/docutils/parsers/rst/states.py", line 325, in section
    self.new_subsection(title, lineno, messages)
  File "/nix/store/mfihnifp4kcydmjasxjmcjwk6pz3lq4k-python3.11-docutils-0.20.1/lib/python3.11/site-packages/docutils/parsers/rst/states.py", line 391, in new_subsection
    newabsoffset = self.nested_parse(
                   ^^^^^^^^^^^^^^^^^^
  File "/nix/store/mfihnifp4kcydmjasxjmcjwk6pz3lq4k-python3.11-docutils-0.20.1/lib/python3.11/site-packages/docutils/parsers/rst/states.py", line 279, in nested_parse
    state_machine.run(block, input_offset, memo=self.memo,
  File "/nix/store/mfihnifp4kcydmjasxjmcjwk6pz3lq4k-python3.11-docutils-0.20.1/lib/python3.11/site-packages/docutils/parsers/rst/states.py", line 195, in run
    results = StateMachineWS.run(self, input_lines, input_offset)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/nix/store/mfihnifp4kcydmjasxjmcjwk6pz3lq4k-python3.11-docutils-0.20.1/lib/python3.11/site-packages/docutils/statemachine.py", line 233, in run
    context, next_state, result = self.check_line(
                                  ^^^^^^^^^^^^^^^^
  File "/nix/store/mfihnifp4kcydmjasxjmcjwk6pz3lq4k-python3.11-docutils-0.20.1/lib/python3.11/site-packages/docutils/statemachine.py", line 445, in check_line
    return method(match, context, next_state)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/nix/store/mfihnifp4kcydmjasxjmcjwk6pz3lq4k-python3.11-docutils-0.20.1/lib/python3.11/site-packages/docutils/parsers/rst/states.py", line 2785, in underline
    self.section(title, source, style, lineno - 1, messages)
  File "/nix/store/mfihnifp4kcydmjasxjmcjwk6pz3lq4k-python3.11-docutils-0.20.1/lib/python3.11/site-packages/docutils/parsers/rst/states.py", line 325, in section
    self.new_subsection(title, lineno, messages)
  File "/nix/store/mfihnifp4kcydmjasxjmcjwk6pz3lq4k-python3.11-docutils-0.20.1/lib/python3.11/site-packages/docutils/parsers/rst/states.py", line 391, in new_subsection
    newabsoffset = self.nested_parse(
                   ^^^^^^^^^^^^^^^^^^
  File "/nix/store/mfihnifp4kcydmjasxjmcjwk6pz3lq4k-python3.11-docutils-0.20.1/lib/python3.11/site-packages/docutils/parsers/rst/states.py", line 279, in nested_parse
    state_machine.run(block, input_offset, memo=self.memo,
  File "/nix/store/mfihnifp4kcydmjasxjmcjwk6pz3lq4k-python3.11-docutils-0.20.1/lib/python3.11/site-packages/docutils/parsers/rst/states.py", line 195, in run
    results = StateMachineWS.run(self, input_lines, input_offset)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/nix/store/mfihnifp4kcydmjasxjmcjwk6pz3lq4k-python3.11-docutils-0.20.1/lib/python3.11/site-packages/docutils/statemachine.py", line 233, in run
    context, next_state, result = self.check_line(
                                  ^^^^^^^^^^^^^^^^
  File "/nix/store/mfihnifp4kcydmjasxjmcjwk6pz3lq4k-python3.11-docutils-0.20.1/lib/python3.11/site-packages/docutils/statemachine.py", line 445, in check_line
    return method(match, context, next_state)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/nix/store/mfihnifp4kcydmjasxjmcjwk6pz3lq4k-python3.11-docutils-0.20.1/lib/python3.11/site-packages/docutils/parsers/rst/states.py", line 2735, in indent
    newline_offset, blank_finish = self.nested_list_parse(
                                   ^^^^^^^^^^^^^^^^^^^^^^^
  File "/nix/store/mfihnifp4kcydmjasxjmcjwk6pz3lq4k-python3.11-docutils-0.20.1/lib/python3.11/site-packages/docutils/parsers/rst/states.py", line 316, in nested_list_parse
    state_machine.run(block, input_offset, memo=self.memo,
  File "/nix/store/mfihnifp4kcydmjasxjmcjwk6pz3lq4k-python3.11-docutils-0.20.1/lib/python3.11/site-packages/docutils/parsers/rst/states.py", line 195, in run
    results = StateMachineWS.run(self, input_lines, input_offset)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/nix/store/mfihnifp4kcydmjasxjmcjwk6pz3lq4k-python3.11-docutils-0.20.1/lib/python3.11/site-packages/docutils/statemachine.py", line 233, in run
    context, next_state, result = self.check_line(
                                  ^^^^^^^^^^^^^^^^
  File "/nix/store/mfihnifp4kcydmjasxjmcjwk6pz3lq4k-python3.11-docutils-0.20.1/lib/python3.11/site-packages/docutils/statemachine.py", line 445, in check_line
    return method(match, context, next_state)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/nix/store/mfihnifp4kcydmjasxjmcjwk6pz3lq4k-python3.11-docutils-0.20.1/lib/python3.11/site-packages/docutils/parsers/rst/states.py", line 2921, in indent
    itemnode, blank_finish = self.definition_list_item(context)
                             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/nix/store/mfihnifp4kcydmjasxjmcjwk6pz3lq4k-python3.11-docutils-0.20.1/lib/python3.11/site-packages/docutils/parsers/rst/states.py", line 2856, in definition_list_item
    self.nested_parse(indented, input_offset=line_offset, node=definition)
  File "/nix/store/mfihnifp4kcydmjasxjmcjwk6pz3lq4k-python3.11-docutils-0.20.1/lib/python3.11/site-packages/docutils/parsers/rst/states.py", line 279, in nested_parse
    state_machine.run(block, input_offset, memo=self.memo,
  File "/nix/store/mfihnifp4kcydmjasxjmcjwk6pz3lq4k-python3.11-docutils-0.20.1/lib/python3.11/site-packages/docutils/parsers/rst/states.py", line 195, in run
    results = StateMachineWS.run(self, input_lines, input_offset)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/nix/store/mfihnifp4kcydmjasxjmcjwk6pz3lq4k-python3.11-docutils-0.20.1/lib/python3.11/site-packages/docutils/statemachine.py", line 233, in run
    context, next_state, result = self.check_line(
                                  ^^^^^^^^^^^^^^^^
  File "/nix/store/mfihnifp4kcydmjasxjmcjwk6pz3lq4k-python3.11-docutils-0.20.1/lib/python3.11/site-packages/docutils/statemachine.py", line 445, in check_line
    return method(match, context, next_state)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/nix/store/mfihnifp4kcydmjasxjmcjwk6pz3lq4k-python3.11-docutils-0.20.1/lib/python3.11/site-packages/docutils/parsers/rst/states.py", line 2357, in explicit_markup
    self.explicit_list(blank_finish)
  File "/nix/store/mfihnifp4kcydmjasxjmcjwk6pz3lq4k-python3.11-docutils-0.20.1/lib/python3.11/site-packages/docutils/parsers/rst/states.py", line 2382, in explicit_list
    newline_offset, blank_finish = self.nested_list_parse(
                                   ^^^^^^^^^^^^^^^^^^^^^^^
  File "/nix/store/mfihnifp4kcydmjasxjmcjwk6pz3lq4k-python3.11-docutils-0.20.1/lib/python3.11/site-packages/docutils/parsers/rst/states.py", line 316, in nested_list_parse
    state_machine.run(block, input_offset, memo=self.memo,
  File "/nix/store/mfihnifp4kcydmjasxjmcjwk6pz3lq4k-python3.11-docutils-0.20.1/lib/python3.11/site-packages/docutils/parsers/rst/states.py", line 195, in run
    results = StateMachineWS.run(self, input_lines, input_offset)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/nix/store/mfihnifp4kcydmjasxjmcjwk6pz3lq4k-python3.11-docutils-0.20.1/lib/python3.11/site-packages/docutils/statemachine.py", line 233, in run
    context, next_state, result = self.check_line(
                                  ^^^^^^^^^^^^^^^^
  File "/nix/store/mfihnifp4kcydmjasxjmcjwk6pz3lq4k-python3.11-docutils-0.20.1/lib/python3.11/site-packages/docutils/statemachine.py", line 445, in check_line
    return method(match, context, next_state)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/nix/store/mfihnifp4kcydmjasxjmcjwk6pz3lq4k-python3.11-docutils-0.20.1/lib/python3.11/site-packages/docutils/parsers/rst/states.py", line 2660, in explicit_markup
    nodelist, blank_finish = self.explicit_construct(match)
                             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/nix/store/mfihnifp4kcydmjasxjmcjwk6pz3lq4k-python3.11-docutils-0.20.1/lib/python3.11/site-packages/docutils/parsers/rst/states.py", line 2367, in explicit_construct
    return method(self, expmatch)
           ^^^^^^^^^^^^^^^^^^^^^^
  File "/nix/store/mfihnifp4kcydmjasxjmcjwk6pz3lq4k-python3.11-docutils-0.20.1/lib/python3.11/site-packages/docutils/parsers/rst/states.py", line 2104, in directive
    return self.run_directive(
           ^^^^^^^^^^^^^^^^^^^
  File "/nix/store/mfihnifp4kcydmjasxjmcjwk6pz3lq4k-python3.11-docutils-0.20.1/lib/python3.11/site-packages/docutils/parsers/rst/states.py", line 2154, in run_directive
    result = directive_instance.run()
             ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/build/Python-3.9.18/Doc/tools/extensions/pyspecific.py", line 361, in run
    label = translators['sphinx'].gettext(label)
            ~~~~~~~~~~~^^^^^^^^^^
KeyError: 'sphinx'

Exception occurred:
  File "/build/Python-3.9.18/Doc/tools/extensions/pyspecific.py", line 361, in run
    label = translators['sphinx'].gettext(label)
            ~~~~~~~~~~~^^^^^^^^^^
KeyError: 'sphinx'
The full traceback has been saved in /build/sphinx-err-k_ryb79y.log, if you want to report the issue to the developers.
Please also report this if it was a user error, so that a better error message can be provided next time.
A bug report can be filed in the tracker at <https://github.com/sphinx-doc/sphinx/issues>. Thanks!
error: builder for '/nix/store/1vij562n5jjgkfnq5a6wka3qkxid2y72-python3.9-3.9.18-doc.drv' failed with exit code 2
error: 1 dependencies of derivation '/nix/store/sci64ads4jnw3d7isag5i13bakl4a3x4-system-path.drv' failed to build
error: 1 dependencies of derivation '/nix/store/i2f2hr5pf1f0hq6kgvvqdxv54agp4d9r-nixos-system-nixos-24.05pre594677.3030f185ba6a.drv' failed to build
```

### Notify maintainers
@fridh@fridh.nl
@FRidh 

### Metadata
Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
[kamilcynk@nixos:~]$  nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-linux"`
 - host os: `Linux 6.6.21, NixOS, 24.05 (Uakari), 24.05pre594677.3030f185ba6a`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.18.1`
 - channels(root): `"nixos"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`
output here
```

**Comment by KCynk** [comment]
Could you check @FRidh ?

**Comment by oliviacrain** [comment]
Reproduces for `python39.doc`, does not reproduce for `python3{10,11,12}.doc`. Given this info and the error itself, I _think_ this will be fixed by backporting python/cpython#98437. I can test this hypothesis and write up a PR in the next few hours.
